ZooProject/src/zoo/Gehege/Aquarium.java

```diff
@@ -24,29 +20,22 @@ public void setTiere(Tier[] tiere) {
 	}
 	
 	public void addTier(Tier tier) {
-		//ein einzelnes Tier wird hinzugefügt zum Gehege
-		//muss auch noch eingeschränkt werden ~> isses nun o/
-		//ich würds so machen
-		if((tier.getTypName()=="Walhai") || (tier.getTypName()=="Kugelfisch")) {
-			//kann zugeordnet werden
-			//referenz: System.out.print("\n " + tier.getTypName());
-			
-			int tiere_array_counter = 0;
-			for(int l= 0; l< this.tiere.length; l++) { 	//schwierig wenn die liste 100
-														//lang ist so wie in Zoo.java
-														//angegeben
-				if(this.tiere[l] != null) {	
-					tiere_array_counter++;
-				}
+		
+		int tiere_array_counter = 0;
+		for(int l= 0; l< this.tiere.length; l++) {
+			if(this.tiere[l] != null) {	
+				tiere_array_counter++;
```

Comment:
Es können in Aquarium alle Tiere zugeordnet werden. Glaub du hast es aus Versehen rausgelöscht, da in Landsaeugetiergehege der if check noch drin steht.

Reply:
Es können nur Wassertiere zugeordnet  werden:
weil alle Wasssertiere den Typen 3 haben... und das Wassertiergehege nur Tiere zulässt mit dem Typ 3

kannst du auch testen indem du folgendes in Main eingibst:
	Adler Igor = new Adler("Igor",sayan_zoo);
	sayan_zoo.addTier(Igor, "Kugelfischtank");

Reply:
hab auch getestet wenn man den Namen aufrufen will vom Gehege in dem Adler Igor lebt kommt ein null Exception Fehler :) -> heißt Igor hat als Gehege null initialisiert

System.out.println(Igor.getGehege().getName());